### PR TITLE
feat(charts): add optional persistence storage to influxdb and grafana

### PR DESCRIPTION
**Pull request by rimusz**

### CODE REVIEW

File: charts/monitor/values.yaml [hunk 1]
```diff
@@ -4,6 +4,11 @@ grafana:
   docker_tag: canary
   # limits_cpu: "100m"
   # limits_memory: "50Mi"
+  persistence:
+    enabled: false
+    storageClass: standard
+    accessMode: ReadWriteOnce
+    size: 5Gi
```
Comment (bacongobbler) [inline charts/monitor/values.yaml:11]: can we please make this configurable?

File: charts/monitor/values.yaml [hunk 2]
```diff
@@ -14,6 +19,11 @@ influxdb:
   database: "kubernetes"
   user: "user"
   password: "password"
+  persistence:
+    enabled: false
+    storageClass: standard
+    accessMode: ReadWriteOnce
+    size: 20Gi
```
Comment (bacongobbler) [inline charts/monitor/values.yaml:26]: can we please make this configurable?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
If you wouldn't mind writing some documentation around how one can get this all set up, that would be great. :)

File: charts/monitor/values.yaml (hunk 2 above)
Comment (rimusz) [reply charts/monitor/values.yaml:26]: I think they are configurable enough already, by default it is disabled.

**Comment by rimusz** [comment]
I was going to write small doc in global `values.yaml` file how to switch persistence on and change other values. Do you want some doc for the website too?

**Comment by bacongobbler** [comment]
yes please.

**Comment by rimusz** [comment]
ok, will do the PR for the global `values.yaml` file, first. when all those bits are merged, will work on doc :)

File: charts/monitor/values.yaml (hunk 2 above)
Comment (bacongobbler) [reply charts/monitor/values.yaml:26]: I'm talking about the actual size of the volume claim. Some users may want smaller or bigger volume claims.

Comment (rimusz) [reply charts/monitor/values.yaml:26]: it is already in `values.yaml` file where user can change. but my idea is put that to global `values.yaml` file too, so enabling persistence you can set the storage size too

Comment (bacongobbler) [reply charts/monitor/values.yaml:26]: sorry! I was clearly misreading it.

**Review by bacongobbler — APPROVED** [review]

**Comment by rimusz** [comment]
will work on global `values.yaml` file changes then

**Comment by rimusz** [comment]
tested the changes of running it in my dev Workflow cluster

**Comment by bacongobbler** [comment]
Closes #93

**Comment by bacongobbler** [comment]
Closes #92

**Review by vdice — APPROVED** [review]
Tested in combination with https://github.com/deis/workflow/pull/699 (PV/PVCs successfully created; grafana dashboards and influxdb metrics persist through pod destruction, as expected.)

**Pull request merged**